datastore/google/cloud/datastore/entity.py

```diff
@@ -144,7 +144,7 @@ def exclude_from_indexes(self):
 
     def __repr__(self):
         if self.key:
-            return '<Entity%s %s>' % (self.key.path,
+            return '<Entity%s %s>' % (self.key._flat_path,
                                       super(Entity, self).__repr__())
         else:
-            return '<Entity %s>' % (super(Entity, self).__repr__())
+            return '<Entity %s>' % (super(Entity, self).__repr__(),)
```

Comment:
If we *are* going to use `repr` for testing, then we need to take hash-order out of the equation.